### PR TITLE
coveralls, badges, etc.

### PR DESCRIPTION
**Pull request by ndushay**
if it's too onerous, I'm happy to break this up into multiple PRs.

take it commit by commit. Each commit should be reasonable.  You may use split view or unified for individual commits.

### CODE REVIEW

File: lib/lyber_utils/file_utilities.rb [hunk 1]
```diff
@@ -109,24 +107,24 @@ def FileUtilities.gpgdecrypt(workspace_dir, targzgpg, targz, passphrase)
...
     def FileUtilities.unpack(original_dir, targz, destination_dir)
- #     LyberCore::Log.debug("unpacking #{targz}")
+      # LyberCore::Log.debug("unpacking #{targz}")
       FileUtils.mkdir_p(destination_dir)
       dir_save = Dir.pwd
       Dir.chdir(destination_dir)
-      unpack_cmd="tar -xzf " + File.join(original_dir, targz)
+      unpack_cmd = "tar -xzf " + File.join(original_dir, targz)
       self.execute(unpack_cmd)
       if not (Dir.entries(destination_dir).length > 0)
```
Comment (tallenaz) [inline lib/lyber_utils/file_utilities.rb]: maybe `unless` here instead of `if not`?

File: lib/lyber_utils/file_utilities.rb [hunk 2]
```diff
@@ -136,34 +134,32 @@ def FileUtilities.unpack(original_dir, targz, destination_dir)
...
+    #     (if nil create sourcename.tar in the same dir as the source object)
+    #
+    # = Return value:
+    # * The method will return true if the transfer is successful.
+    # * The method will raise an exception if either the rsync command fails,
+    # or a test for the existence of the transferred object fails.
+    # The exception's message will contain the explanation of the failure
+    #
+    def FileUtilities.tar_object(source_path, dest_path=nil)
+      if (dest_path.nil?)
```
Comment (tallenaz) [inline lib/lyber_utils/file_utilities.rb]: Unnecessary parens?

File: lyber-utils.gemspec [hunk 3]
```diff
@@ -1,30 +1,30 @@
 # -*- encoding: utf-8 -*-
 lib = File.expand_path('../lib/', __FILE__)
 $:.unshift lib unless $:.include?(lib)
-  
+
 Gem::Specification.new do |s|
   s.name        = "lyber-utils"
   s.version     = "0.1.3"
   s.platform    = Gem::Platform::RUBY
   s.authors     = ["Richard Anderson","Willy Mene","Michael Klein"]
```
Comment (tallenaz) [inline lyber-utils.gemspec]: Add spaces between array elements?

File: lib/lyber_utils/checksum_validate.rb [hunk 4]
```diff
@@ -77,33 +76,31 @@ def self.md5_hash_from_content_metadata(content_md)
...
+    # * The method will return true if the verification is successful.
+    # * The method will raise an exception if either the md5sum command fails,
+    # or a test of the md5sum output indicates a checksum mismatch.
+    # The exception's message will contain the explaination of the failure.
     def self.verify_md5sum_checksums(directory, checksum_file)
-#      LyberCore::Log.debug("verifying checksums in #{directory}")
+      # LyberCore::Log.debug("verifying checksums in #{directory}")
       dir_save = Dir.pwd
       Dir.chdir(directory)
       checksum_cmd = 'md5sum -c ' + checksum_file + ' | grep -v OK | wc -l'
```
Comment (tallenaz) [inline lib/lyber_utils/checksum_validate.rb]: Interpolation instead of `+`?

File: lib/lyber_utils/checksum_validate.rb [hunk 5]
```diff
@@ -77,33 +76,31 @@ def self.md5_hash_from_content_metadata(content_md)
...
       badcount = FileUtilities.execute(checksum_cmd).to_i
-      if not (badcount==0)
+      if not (badcount == 0)
```
Comment (tallenaz) [inline lib/lyber_utils/checksum_validate.rb]: Unnecessary parens?

File: lib/lyber_utils/file_utilities.rb [hunk 6]
```diff
@@ -25,80 +24,79 @@ def FileUtilities.execute(command)
...
+    # "pair tree" hierarchical structure, where the path is derived
+    # from segments of a barcode string
+    #
+    # = Input:
+    # * barcode = barcode string
+    #
+    # = Return value:
+    # * A string containing a slash-delimited dirname derived from the barcode
     def FileUtilities.pair_tree_from_barcode(barcode)
       if (barcode.class != String)
```
Comment (tallenaz) [inline lib/lyber_utils/file_utilities.rb]: Unnecessary parens? I'm seeing lots around, so will stop harping on it :)

File: lib/lyber_utils/file_utilities.rb [hunk 7]
```diff
@@ -25,80 +24,79 @@ def FileUtilities.execute(command)
...
+    # The exception's message will contain the explanation of the failure
+    #
+    # Network transfers will only succeed if the appropriate public key
+    # authentication has been previously set up.
     def FileUtilities.transfer_object(filename, source_dir, dest_dir)
-      source_path=File.join(source_dir, filename)
-      rsync='rsync -a -e ssh '
+      source_path = File.join(source_dir, filename)
+      rsync = 'rsync -a -e ssh '
       rsync_cmd = rsync + "'" + source_path + "' " + dest_dir
```
Comment (tallenaz) [inline lib/lyber_utils/file_utilities.rb]: Again, (here and elsewhere) interpolation instead of `+`?

File: lib/lyber_utils/file_utilities.rb [hunk 8]
```diff
@@ -109,24 +107,24 @@ def FileUtilities.gpgdecrypt(workspace_dir, targzgpg, targz, passphrase)
       return true
```
Comment (tallenaz) [inline lib/lyber_utils/file_utilities.rb]: Arguably a needless use of `return`, maybe just `true`?

**Comment by ndushay** [comment]
@tallenaz  There are so many style issues that rubocop noticed that I didn't try to fix them all.  This particular gem is used elsewhere, so I deemed it worth it to fix all the particular things you noticed, and a few more.   So I have revised this PR and it is ready for re-review.

I guess the standard for this is merge if:
1. changes are not harmful 
2. at least some changes are an improvement;  all are improvements or benign.

**Pull request merged**